docs/configuration/configuration.md

```diff
@@ -121,6 +121,10 @@ global:
   # that will be kept in memory. 0 means no limit.
   [ keep_dropped_targets: <int> | default = 0 ]
 
+  # Configure the Go garbage collector [GOGC parameter](https://tip.golang.org/doc/gc-guide#GOGC)
+  # Lowering this number increases CPU usage.
+  [ gogc: <int> | default = 50 ]
```

Comment:
because we provide `auto-gomemlimit.ratio` as a CL argument (and gomaxprocs auto setting as arguments as well), I think it's better to have gogc as a arg as well (all config in the same place)

Reply:
I would actually like to migrate those flags to the config file. This is mostly my first attempt at adding such a runtime tuneable to the config.

Reply:
I added the `gogc` entry to `global` as there was already a struct for it. But maybe we should have a `runtime` top-level config key? Not sure what other opinions are about that.

Reply:
I see, we can keep it for now I think, and group it with the others when they're migrated.

Reply:
Well, I'd rather avoid moving config keys.

So the question is, do we want to keep it in `global`, add `global.runtime`, or add `runtime` as a top-level key. Maybe some other folks can weigh in on this.

Reply:
I misread, I thoughts you were talking about the internal structs.
+1 for `runtime` as a top-level key.

Reply:
Slack conv: https://cloud-native.slack.com/archives/C05U6G6T8G7/p1717487320757739

+1 for runtime

Reply:
weak +1 for `global.runtime`.

`global` already carries fields unrelated to the scrape configs (like `query_log_file`) and the `runtime`  sub-entry makes it easier to reason about.